impl/src/main/java/org/ehcache/impl/internal/store/DefaultStoreProvider.java

```diff
@@ -46,10 +49,22 @@
 public class DefaultStoreProvider implements Store.Provider {
   private static final Logger LOG = LoggerFactory.getLogger(DefaultStoreProvider.class);
 
+  private static final Set<ResourceType> SUPPORTED_RESOURCE_TYPES =
+      new HashSet<ResourceType>(Arrays.asList(ResourceType.Core.HEAP, ResourceType.Core.OFFHEAP, ResourceType.Core.DISK));
+
   private volatile ServiceProvider<Service> serviceProvider;
   private final ConcurrentMap<Store<?, ?>, Store.Provider> providersMap = new ConcurrentWeakIdentityHashMap<Store<?, ?>, Store.Provider>();
 
   @Override
+  public int rank(final Set<ResourceType> resourceTypes, final Collection<ServiceConfiguration<?>> serviceConfigs) {
+    if (SUPPORTED_RESOURCE_TYPES.containsAll(resourceTypes)) {
+      return resourceTypes.size();
```

Comment:
This is valid only if bigger than `1`, otherwise it may get selected and later fail.